src/main/kotlin/ecommerce/interceptor/AuthInterceptor.kt

```diff
@@ -0,0 +1,106 @@
+package ecommerce.interceptor
+
+import ecommerce.service.TokenService
+import jakarta.servlet.http.HttpServletRequest
+import jakarta.servlet.http.HttpServletResponse
+import org.springframework.stereotype.Component
+import org.springframework.web.servlet.HandlerInterceptor
+
+@Component
+class AuthInterceptor(private val tokenService: TokenService) : HandlerInterceptor {
+    companion object {
+        const val USER_ID_ATTRIBUTE = "userId"
+        const val USER_ROLE_ATTRIBUTE = "userRole"
+        const val USER_EMAIL_ATTRIBUTE = "userEmail"
+        const val USER_NAME_ATTRIBUTE = "userName"
+        private const val BEARER_PREFIX = "Bearer "
+        private const val ADMIN_PATH_PREFIX = "/admin/"
+        private const val ADMIN_ROLE = "ADMIN"
+    }
```

Comment:
In Kotlin, `companion object` is usually placed at the bottom. Check the convention through the link below. 🙂
https://kotlinlang.org/docs/coding-conventions.html#class-layout